src/core/selection/selection.view.js

```diff
@@ -2,43 +2,243 @@ import View from '../view/view';
 import Command from 'core/infrastructure/command';
 import behaviorFactory from './selection.factory';
```

Comment:
This comment is not about this PR. Maybe we should have `strategyFactory` instead of `behaviorFactory`?
Strategy is a common abstraction to encapsulate behavior